lib/led.js

```diff
@@ -23,27 +143,36 @@ function Led(opts) {
   var pinValue;
   var defaultLed;
   var state;
-  var isFirmata = true;
 
   if (!(this instanceof Led)) {
     return new Led(opts);
   }
 
+  var controller = null;
+  var err = null;
+
+  if (!opts || !opts.hasOwnProperty("controller")) {
+    controller = Controllers["Standard"];
+  }
+  else if (typeof opts.controller === "string") {
+    controller = Controllers[opts.controller];
+  }
+  else {
+    controller = opts.controller;
+  }
+
+  if (controller == null) {
+    controller = Controllers["Standard"];
+  }
+
+  Object.defineProperties(this, controller);
```

Comment:
This seems unnecessary since opts alway exists either as a number or an object. If opts does not exist we don't have a pin number so the instance is doomed anyway.

Are their reasons for doing it this way that I don't see?

It's handled like this in Servo and Motor:
````
if (this.opts.controller) {
  controller = typeof this.opts.controller === "string" ?
    Controllers[this.opts.controller] : this.opts.controller;

  Object.defineProperties(this, controller);
}
````

Is that something we should change?